kubectl-volsync/cmd/migration_create.go

```diff
@@ -72,34 +108,294 @@ func validateMigrationCreate(cmd *cobra.Command, args []string) error {
+	}
+
+	serviceType, err := cmd.Flags().GetString("servicetype")
+	if err != nil {
+		return fmt.Errorf("please provide service type, err = %w", err)
+	}
+
+	if v1.ServiceType(serviceType) != v1.ServiceTypeClusterIP &&
+		v1.ServiceType(serviceType) != v1.ServiceTypeLoadBalancer {
+		return fmt.Errorf("unsupported service type: %v", v1.ServiceType(serviceType))
+	}
+	mrd.Destination.ServiceType = (*v1.ServiceType)(&serviceType)
+	mrd.MDName = mrd.Namespace + "-" + mrd.PVCName + "-migration-dest"
+
+	return nil
+}
+
+func (mc *migrationCreate) Run(ctx context.Context) error {
+	// create namespace if does not exist
+	_, err := mc.createNamespace(ctx)
 	if err != nil {
 		return err
 	}
-	relation, err := createRelationship(configDir, rName, MigrationRelationship)
+	// create destination PVC if does not exist
+	if mc.PVC == nil {
+		mc.PVC, err = mc.createDestinationPVC(ctx)
+		if err != nil {
+			return err
+		}
+	}
+	// create migration destination
+	err = mc.createDestination(ctx)
 	if err != nil {
 		return err
 	}
+	// save the destination details into config file
+	if err = mc.mr.Save(); err != nil {
+		return fmt.Errorf("unable to save relationship configuration: %w", err)
+	}
+	return nil
+}
 
-	// Insert information into the relationship & save it
-	cap, err := cmd.Flags().GetString("capacity")
-	if err != nil {
+func (mc *migrationCreate) createNamespace(ctx context.Context) (*v1.Namespace, error) {
+	mrd := mc.mr.data.Destination
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: mrd.Namespace,
+		},
+	}
+	if err := mc.clientObject.Create(ctx, ns); err != nil {
+		if kerrs.IsAlreadyExists(err) {
+			klog.Infof("Namespace: \"%s\" already present, proceeding with this namespace",
+				mrd.Namespace)
+			return ns, nil
+		}
+		return nil, err
+	}
+	klog.Infof("Created destination namespace: \"%s\"", mrd.Namespace)
+
+	return ns, nil
+}
+
+func (mc *migrationCreate) createDestinationPVC(ctx context.Context) (*v1.PersistentVolumeClaim, error) {
+	mrd := mc.mr.data.Destination
+	var destPVC *v1.PersistentVolumeClaim
+
+	if *mrd.Destination.StorageClassName == "" {
+		destPVC = &v1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      mrd.PVCName,
+				Namespace: mrd.Namespace,
+			},
+			Spec: v1.PersistentVolumeClaimSpec{
+				AccessModes: mrd.Destination.AccessModes,
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceStorage: *mrd.Destination.Capacity,
+					},
+				},
+			},
+		}
+	} else {
+		destPVC = &v1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      mrd.PVCName,
+				Namespace: mrd.Namespace,
+			},
+			Spec: v1.PersistentVolumeClaimSpec{
+				AccessModes:      mrd.Destination.AccessModes,
+				StorageClassName: mrd.Destination.StorageClassName,
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceStorage: *mrd.Destination.Capacity,
+					},
+				},
+			},
+		}
+	}
```

Comment:
A couple things about this block...

While we're not making the distinction (currently) between an empty StorageClassName (=="") and a nil one (==nil) on the cli, they have different meanings to Kubernetes. Nil means "use the default", while empty means "use one with an empty name". When we parse the cli, we should put the parameter into a `*string` so that we can preserve that distinction.

After parsing the cli arg into the `*string`, the duplication here can be removed.

Reply:
sure. will set `StorageClassName = nil` if arg is coming empty